### DISCUSSION TITLE
Preventing Text Overflow when a Merged Cell Contains Page Break

### DISCUSSION THREAD

**Discussion by michael-mac** [eclipse-birt/birt#1333] [General]
I was hoping for help on a particularly specific issue I'm having with a BIRT report.

The structure of the problematic section of the report is 4 rows by 5 columns, within a Table. The leftmost column has one single cell, merged across all 4 rows, that prints a lot of text. It is possible that the page can break only between the third row and fourth row - the first 3 rows have the needed "Avoids" on page break to keep them together, but we have "Auto" between rows 3+4.

**Worst case, I can update the report to Avoid page breaks between rows 3 and 4, but I would prefer not to do this.**

The problem is that the leftmost column's text will often be longer than the first 3 rows' height when there **is** a page break. This will cause the text to spill out of the left cell at the bottom of the page - even though there is plenty of room for the row to continue to encapsulate the text, it will not do so (I suspect the reason for this is that the report sees we still have the fourth row "technically available" to contain that text despite the page break since it is merged, so it does not auto-extend the first 3 rows to fix it).

![image](https://github.com/eclipse-birt/birt/assets/34662846/137070fb-1db8-47ff-8fce-3f3c9c4f4eeb)

I need to keep the text itself Avoiding the page break, so it cannot spill over onto the next page. The ideal result is that all the text is within the merged cell on the first page, so the first 3 rows extend as needed, and the second page just has the rest of the cell blank, almost appearing as if the leftmost cell has actually split between the pages.

I was thinking of creating a variable that somehow identifies when we will have a page break between the 3rd and 4th row (this is the hardest part in my mind - I don't think onPageBreak scripts will work, will they?), that can then do 1 of 2 things:
- equally auto-extend the height of the first 3 rows to add exactly the height needed to contain the "spillover" text, or
- use Visibilities to either split the 4th row or don't, depending on the variable state

Is it possible to accomplish this?

**Comment by hvbtup** [comment]
What BIRT release are you using?
Do you mean PDF output?
What are the emitter options?

And you say "Table". Do you mean a Table Item? And are all the rows detail rows?

It would be best if you could provide a minimal example demonstrating the issue.
